include/xtensor/xexpression.hpp

```diff
@@ -302,6 +318,135 @@ namespace xt
                                                   >
     {
     };
+
+#define FORWARD_METHOD(name)                   \
+    auto name() const                          \
+        -> decltype(std::declval<E>().name())  \
+    {                                          \
+        return m_ptr->name();                  \
+    }
```

Comment:
I wonder if we can have an approach for this, that supports varying overloads for lvalues and ravlues.